src/Attachment.tsx

```diff
@@ -25,6 +71,10 @@ const Media = (props: {
     const { type, ... mediaProps } = props; // this allows us to keep 'type' out of the final HTML
     switch (type) {
         case 'video':
+            if (isYoutube(mediaProps.src)) {
+                return <Youtube { ... mediaProps } />
+            }
+
```

Comment:
It bugs me that we end up calling URI twice on the same string. And if we add future special cases, we'll do it more.

How about we turn this into a function called Video which will look at the URI for each special-case domain in turn, returning the appropriate `<iframe>` as appropriate, or returning a `<video>` tag if none apply. All that logic can be in one function.